src/algebra/integrator.hpp

```diff
@@ -144,13 +165,32 @@ Integrator<T>::Integrator(MODEL::Model<T> &model, T time_step_size, T rho,
   system.reserve(model);
 }
 
+template <typename T>
+Integrator<T>::Integrator() {}
+
 template <typename T>
 Integrator<T>::~Integrator() {}
 
+template <typename T>
+void Integrator<T>::clean() {
+  // Cannot be in destructor because dynamically allocated pointers will be lost
+  // when objects are assigned from temporary objects.
+  system.clean();
+}
+
+template <typename T>
+void Integrator<T>::update_params(MODEL::Model<T> &model, T time_step_size) {
+  y_init_coeff = alpha_f * 0.5 * time_step_size;
+  this->time_step_size = time_step_size;
+  time_step_size_inv = 1.0 / time_step_size;
+  y_dot_coeff = alpha_m * alpha_f_inv * gamma_inv * time_step_size_inv;
+  model.update_constant(system);
+  model.update_time(system, 0.0);
+}
+
```

Comment:
Would it make sense to call `update_params` from the constructor to avoid code redundancy?